src/routes/safe/components/Apps/components/AddAppForm/index.tsx

```diff
@@ -39,38 +40,34 @@ const INITIAL_VALUES: AddAppFormValues = {
   agreementAccepted: false,
 }
 
-const APP_INFO: SafeApp = {
-  id: '',
-  url: '',
-  name: '',
-  iconUrl: appsIconSvg,
-  error: false,
-  description: '',
-}
+const APP_INFO = getEmptySafeApp()
 
 interface AddAppProps {
   appList: SafeApp[]
   closeModal: () => void
-  formId: string
-  onAppAdded: (app: SafeApp) => void
-  setIsSubmitDisabled: (disabled: boolean) => void
 }
 
-const AddApp = ({ appList, closeModal, formId, onAppAdded, setIsSubmitDisabled }: AddAppProps): React.ReactElement => {
-  const [appInfo, setAppInfo] = React.useState<SafeApp>(APP_INFO)
+const AddApp = ({ appList, closeModal }: AddAppProps): ReactElement => {
+  const [appInfo, setAppInfo] = useState<SafeApp>(APP_INFO)
+  const history = useHistory()
+  const matchSafeWithAddress = useRouteMatch<{ safeAddress: string }>({ path: `${SAFELIST_ADDRESS}/:safeAddress` })
 
   const handleSubmit = () => {
-    closeModal()
-    onAppAdded(appInfo)
+    const newAppList = [
+      { url: appInfo.url, disabled: false },
+      ...appList.map(({ url, disabled }) => ({ url, disabled })),
+    ]
+    saveToStorage(APPS_STORAGE_KEY, newAppList)
+    const goToApp = `${matchSafeWithAddress?.url}/apps?appUrl=${encodeURI(appInfo.url)}`
+    history.push(goToApp)
   }
```

Comment:
This only updates the stored app list and not app's state, is it intended?

Reply:
sorry, what do you mean by app's state?

Reply:
we hold an apps list in state inside useAppList hook

Reply:
ahh, I see what you mean. No, it's not needed anymore, because the useAppList is only used in /apps URL, when we add a new app, we redirect it to /apps/new_app_url. 
then, if the user goes back to /apps it will load the just added_app because it's already in the storage.